### PR TITLE
Upgrades algoliasearch javascript package

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
Upgrades `algoliasearch` JS package from v3 to v4, updates some code accordingly, and a small refactor of some code in `search.js`

### Motivation
Previous version was way out of date, this will also help make upcoming algolia insights work easier.

### Preview
https://docs-staging.datadoghq.com/brian.deutsch/upgrade-algoliasearch/search/?s=synthetics
Search page continues to work as expected, including pagination.

https://docs-staging.datadoghq.com/brian.deutsch/upgrade-algoliasearch/
Searching from the home page and the sidebar continues to work as expected, including using the autocomplete dropdown.

Search results themselves should be unaffected, and there should be no console errors.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: src/scripts/components/search.js [hunk 1]
```diff
@@ -1,436 +1,426 @@
...
+
+const renderResults = (query, hits) => {
+    $('#tipue_search_input').val(query);
+    appendSearchResultCountString(hits.length);
+    let formatted_results = '';
+
+    if (hits.length) {
+        for (const i in hits) {
+            const hit = hits[i];
+            formatted_results += '<div class="hit">';
```
Comment (bgdeutsch) [inline src/scripts/components/search.js:56]: actual rendering of results is legacy code, i have a Jira ticket for it here https://datadoghq.atlassian.net/browse/WEB-1928.    will address this in a later PR.

File: .gitignore [hunk 2]
```diff
@@ -3,6 +3,7 @@ public/
 data/service_checks/
 data/npm
 integrations_data
+.hugo_build.lock
```
Comment (davidejones) [inline .gitignore:6]: I'm not that familiar with this new file I'm wondering if it should be committed instead of ignored? like yarn lock?

**Review by davidejones — APPROVED** [review]
Other than my comment, everything looks good

File: .gitignore (hunk 2 above)
Comment (bgdeutsch) [reply .gitignore:6]: @davidejones it looks like this was introduced when we recently upgraded Hugo versions.  The release notes recommend to leave the file alone, and .gitignore if we dont want it in the repository: https://gohugo.io/news/0.89.0-relnotes/#notes

Comment (nsollecito) [reply .gitignore:6]: @davidejones this is a new lock file as part of the latest Hugo version. It has been suggested to never commit this file -- https://github.com/unicef/inventory/commit/5f978c252202dedb6c97782380df9b6f6d7c56e1

Comment (davidejones) [reply .gitignore:6]: ok got it thanks!

**Review by ruthnaebeck — COMMENTED** [review]
If we are ignoring the `.hugo_build.lock` file, shouldn't that file be deleted from this PR?

**Review by ruthnaebeck — APPROVED** [review]
lgtm

**Pull request merged**